src/h.ts

```diff
@@ -1,6 +1,6 @@
 import {vnode, VNode, VNodeData} from './vnode';
 export type VNodes = Array<VNode>;
-export type VNodeChildElement = VNode | string | number | undefined | null;
+export type VNodeChildElement = VNode | string | number | boolean | undefined | null;
```

Comment:
Is this addition of `boolean` a leftover from when JSX support was added  directly to`h`?

Reply:
Would you be okay if we converted all primitives to string when making vnodes text properties. This would simplify a bunch of things and not have special children type for jsx vs h.

I think @mightyiam's decision behind jsx being separate is that he wanted flatChildren not to impact h calls since that may be a surface area change for h  and possibly a small perf hit.

Adding `boolean`  was going back to the original comment you made.

```
I definitely see your point. On the other hand, the childen type is already fairly liberal and the only extra thing that gets added is boolean.

Could another option be to not add boolean to the VNodeChildElement type but still support it in the code? I don't think the TypeScript JSX mechanism would care about that so it would give us the best of both worlds. h would still have the stricter type for the benefit of TypeScript developers and it will work with JSX.
```